File: python/Two_Sum.py
nums = [2, 7, 11, 15]
target = 9

class Solution:
    def two_sum(self, nums: list[int], target: list[int]) -> list[int]:
        num_indices = {}
        for i, num in enumerate(nums):
            complement = target - num 
            if complement in num_indices:
                return [num_indices[complement], i]
            num_indices[num] = i
        return []

solution = Solution()
result = solution.two_sum(nums, target)
print(f"Indices dos números que somam {target}: {result}")

File: rust/Two_Sum/Two_Sum.rs
use std::collections::HashMap;

fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut num_indices: HashMap<i32, i32> = HashMap::new();

    for (index, &num) in nums.iter().enumerate() {
        let complement = target - num; //Procurando o numero que somado complementa o target

        if let Some(&complement_index) = num_indices.get(&complement) {
            return vec![complement_index, index as i32];
        }

        num_indices.insert(num, index as i32);
    }

    // Retorna um vetor vazio se não encontrar uma solução
    vec![]
}

fn main() {
    let nums = vec![2, 7, 11, 15];
    let target = 9;

    let result = two_sum(nums, target);
    println!("Indices dos números que somam {}: {:?}", target, result);
}

